### PR TITLE
Add live data to Shared preferences functionality.

### PR DESCRIPTION
**Pull request by SofiaZarzabal**
## Description
Add live data functionality to Shared Preference screen.
Also add some visual fixes and change text from default value constant when no name is saved.

### Evidence

https://github.com/SofiaZarzabal/DataStoreApp/assets/84337427/e983634c-54ed-4e77-ba4d-f7f656851929

### CODE REVIEW

File: app/src/main/java/com/example/datastoreapp/activity/MainActivity.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ package com.example.datastoreapp.activity
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import com.example.datastoreapp.databinding.ActivityMainBinding
-
+// Borrar esta activity y su layout
```
Comment (FranciscoGaraventa) [inline app/src/main/java/com/example/datastoreapp/activity/MainActivity.kt:6]: This is a WIP? Did you forget to remove this comment?

File: app/src/main/java/com/example/datastoreapp/viewmodel/SharedPreferencesViewModel.kt [hunk 2]
```diff
@@ -1,13 +1,22 @@
 package com.example.datastoreapp.viewmodel
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.example.datastoreapp.storetype.SharedPreferencesStore
 
-class SharedPreferencesViewModel(private val sharedPreferencesStorage: SharedPreferencesStore) : ViewModel() {
+class SharedPreferencesViewModel(private val sharedPreferencesStorage: SharedPreferencesStore) :
```
Comment (FranciscoGaraventa) [inline app/src/main/java/com/example/datastoreapp/viewmodel/SharedPreferencesViewModel.kt:8]: You are going to have one ViewModel defined per each screen/type of store?

File: app/src/main/java/com/example/datastoreapp/activity/MainActivity.kt (hunk 1 above)
Comment (SartiMau) [reply app/src/main/java/com/example/datastoreapp/activity/MainActivity.kt:6]: +1

File: app/src/main/java/com/example/datastoreapp/viewmodel/SharedPreferencesViewModel.kt (hunk 2 above)
Comment (SartiMau) [reply app/src/main/java/com/example/datastoreapp/viewmodel/SharedPreferencesViewModel.kt:8]: I think she will need it to manage the different storages

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Pull request closed without merging**